app/controllers/media_controller.rb

```diff
@@ -148,4 +158,14 @@ def redirect_to_current_filename
   def temporary_redirect?
     false
   end
+
+  def set_token_payload
+    token = params.fetch(:token, cookies[:auth_bypass_token])
+    @token_payload = if token
+                       secret = Rails.application.secrets.jwt_auth_secret
+                       JWT.decode(token, secret, true, algorithm: "HS256").first
```

Comment:
Wondering why `.first` is needed here 🤔 `JWT.decode` takes a token and returns the unencrypted payload, not an array...?

Reply:
If I remove it, I'm getting: 🤷‍♀ 
```
asset.valid_auth_bypass_token?(@token_payload["sub"])
no implicit conversion of String into Integer
```
and the previous version was also taking the first element of an array by adding the comma after payload 
https://github.com/alphagov/asset-manager/blob/afcd1f5b0f05a5ad42f11d208d178c06c494c655/app/models/asset.rb#L98

Reply:
🤢 well `.first` is much better than an easy to miss `,` 😄

Reply:
🤢 well `.first` is much better than an easy to miss `,` 😄